src/nn/ops/binary.ts

```diff
@@ -15,6 +16,34 @@ export abstract class Binary extends SingleOutputOperation {
     this.a_ = a;
     utils.validateOperand(b);
     this.b_ = b;
+    this.createOutput();
+  }
+
+  createOutput(): void {
+    if (this instanceof MatMul) {
+      const rankA = this.a_.rank();
+      const rankB = this.b_.rank();
+      if (rankA === 1 && rankB === 1) {
+        this.outputShape_ = []; // scalar
+      } else if (rankA >= 2 && rankB === 1) {
+        this.outputShape_ = this.a_.shape().slice();
+        this.outputShape_[rankA - 1] = 1;
+      } else if (rankA === 1 && rankB >= 2) {
+        this.outputShape_ = this.b_.shape().slice();
+        this.outputShape_[rankB - 2] = 1;
```

Comment:
You may want to remove the 1-D tensor support. The spec now requires inputs at least to be 2-D tensors.  https://www.w3.org/TR/webnn/#api-mlgraphbuilder-matmul